### PR TITLE
feat(connector): [Payone] add connector template code

### PR DESCRIPTION
**Pull request by KiranKBR**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Template code for Payone Payout

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [x] This PR modifies application configuration/environment variables

## Motivation and Context
https://github.com/juspay/hyperswitch-cloud/issues/4902

## How did you test it?
This is template code so, no api testing required  

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/payments/flows.rs [hunk 1]
```diff
@@ -142,6 +142,7 @@ impl<const T: u8>
 }
 
 default_imp_for_complete_authorize!(
+    connector::Payone,
```
Comment (srujanchikke) [inline crates/router/src/core/payments/flows.rs:145]: This should be added in alphabetical order

File: crates/router/src/core/payments/flows.rs [hunk 2]
```diff
@@ -2574,6 +2602,7 @@ impl<const T: u8>
 {
 }
 default_imp_for_connector_authentication!(
+    connector::Payone,
```
Comment (srujanchikke) [inline crates/router/src/core/payments/flows.rs:2605]: Please add the connector enum in alphabetical order in all other places.

File: crates/router/src/core/payments/flows.rs (hunk 1 above)
Comment (KiranKBR) [reply crates/router/src/core/payments/flows.rs]: done

File: crates/router/src/core/payments/flows.rs (hunk 2 above)
Comment (KiranKBR) [reply crates/router/src/core/payments/flows.rs]: done

**Review by srujanchikke — APPROVED** [review]

**Review by SamraatBansal — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**